core_lib/src/interface/backupelement.h

```diff
@@ -18,76 +19,142 @@ GNU General Public License for more details.
 #ifndef BACKUPELEMENT_H
 #define BACKUPELEMENT_H
 
-#include <QObject>
-#include "vectorimage.h"
+#include <QUndoCommand>
+#include <QRectF>
+
 #include "bitmapimage.h"
-#include "soundclip.h"
+#include "vectorimage.h"
 
 class Editor;
-
-class BackupElement : public QObject
+class BackupManager;
+class PreferenceManager;
+class SoundClip;
+class Camera;
+class Layer;
+class KeyFrame;
+class TransformElement;
+
+class BackupElement : public QUndoCommand
 {
-    Q_OBJECT
 public:
-    enum types { UNDEFINED, BITMAP_MODIF, VECTOR_MODIF, SOUND_MODIF };
-
-    QString undoText;
-    bool somethingSelected = false;
-    qreal rotationAngle = 0.0;
-    qreal scaleX = 1.0;
-    qreal scaleY = 1.0;
-    QPointF translation;
-    QRectF mySelection;
-    QPointF selectionAnchor;
-
-    virtual int type() { return UNDEFINED; }
-    virtual void restore(Editor*) { Q_ASSERT(false); }
+    explicit BackupElement(Editor* editor, QUndoCommand* parent = nullptr);
```

Comment:
`QUndoCommand* parent = nullptr` this parameter is not used and can be removed?

Reply:
BackupElement does not use it itself, but it's derived classes do. 
It has to be passed to QUndoCommand at the end and since we pass through to BackupElement, it is needed?

Reply:
Ah, I see it now. A BitmapElement is created and then followed by a TransformElement, which uses the BitmapElement as its parent. Thanks.